File: src/main/java/kz/smarthealth/patientservice/repository/PatientRepository.java
package kz.smarthealth.patientservice.repository;

import kz.smarthealth.patientservice.model.entity.PatientDocument;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PatientRepository extends MongoRepository<PatientDocument, String> {

    List<PatientDocument> findAllByUserId(String userId);
}

File: src/main/kotlin/kz/samat/patientservice/repository/PatientRepository.kt
package kz.samat.patientservice.repository

import kz.samat.patientservice.model.Patient
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

/**
 * Repository for [Patient]
 *
 * Created by Samat Abibulla 2022/12/13
 */
@Repository
interface PatientRepository : MongoRepository<Patient, String> {

    /**
     * Returns set of patients by user id
     *
     * @param userId id of user that own patients
     * @return set of patients
     */
    fun findAllByUserId(userId: String): List<Patient>
}